### ISSUE TITLE
[FEATURE] Remove SDKClusterSettings update method taking a map 

### ISSUE THREAD

**Issue by dbwiddis** [opensearch-project/opensearch-sdk-java#512]
### Is your feature request related to a problem?

A settings update consumer method taking a map was part of the original update design (#138) and has survived multiple refactorings (#352, #389, #435, #443).  Most recently in #510 we had to create an internal map to store pending updates, sent in batch after initialization.  This accomplishes the "efficiency" purpose of this constructor.

Since the primary use case of these updates is in extension actions which are instantiate via injection before ExtensionsRunner initialization,  and because we want to keep the same API to minimize the diff, and since creating the map involves a lot of class casting, it's somewhat smelly and no longer provides added value.

### What solution would you like?

Remove the `SDKClusterSettings` `addSettingsUpdateConsumer(Map<Setting<?>, Consumer<?>> settingsUpdateConsumers)` method (and its associated tests).  

Refactor the existing AD extension uses of this map to restore the original code calling the `addSettingsUpdateConsumer(Setting<T> setting, Consumer<T> settingsUpdateConsumer)` method.  (Copy/paste those updates from the `main` branch is probably the easiest path to doing this.)

### What alternatives have you considered?

Leaving it as is to provide multiple APIs.  This causes confusion and isn't really needed any more.

**Comment by dbwiddis** [comment]
Since both methods are available now, it would be useful to complete the AD refactoring to the single method first, before removing this.

**Comment by qchimera** [comment]
> AD refactoring

Hi! Wanna take this issue, what does the "AD" mean?

**Comment by dbwiddis** [comment]
> Hi! Wanna take this issue, what does the "AD" mean?

Looks like I got this answered in your PR before seeing your question on this issue. :-)

AD is the Anomaly Detection project.   We are migrating that plugin code to extension code in the `feature/extensions` branch.   In the PR I posted a screencap of the three classes there (on `feature/extensions` branch) that need to be changed back to the original format that you'll find on the `main` branch.
